isthmus/src/main/java/io/substrait/isthmus/SqlToSubstrait.java

```diff
@@ -84,36 +116,35 @@ private Plan executeInner(
       SqlValidator validator,
       CalciteCatalogReader catalogReader)
       throws SqlParseException {
-    RelRoot root = sqlToRelNode(sql, factory, validator, catalogReader);
-
-    // System.out.println(RelOptUtil.toString(root.rel));
-    Rel pojoRel = SubstraitRelVisitor.convert(root, EXTENSION_COLLECTION);
-    FunctionCollector functionCollector = new FunctionCollector();
-    RelProtoConverter toProtoRel = new RelProtoConverter(functionCollector);
-    var protoRel = pojoRel.accept(toProtoRel);
-
-    var planRel =
-        PlanRel.newBuilder()
-            .setRoot(
-                io.substrait.proto.RelRoot.newBuilder()
-                    .setInput(protoRel)
-                    .addAllNames(TypeConverter.toNamedStruct(root.validatedRowType).names()));
-
     var plan = Plan.newBuilder();
-    plan.addRelations(planRel);
+    FunctionCollector functionCollector = new FunctionCollector();
+    var relProtoConverter = new RelProtoConverter(functionCollector);
+    sqlToRelNode(sql, validator, catalogReader)
+        .forEach(
+            root -> {
+              plan.addRelations(
+                  PlanRel.newBuilder()
+                      .setRoot(
+                          io.substrait.proto.RelRoot.newBuilder()
+                              .setInput(
+                                  SubstraitRelVisitor.convert(root, EXTENSION_COLLECTION)
```

Comment:
Here we use "one fails all" policy. That is, failure of one query's conversion would fail all the queries. This is fine for executing a transaction with multi-queries. In other scenarios, we may want to keep the successful conversions, and record the failed queries in result.  Probably something to consider in the future.

Reply:
Good point. Added TODO here.